Feature/series

skaik-mo opened the pull request:
implement series domain module to provide series features.
implements series repository to handle multiple data sources for the series (cache, remote api).
implement remote series data source using TMDP api.
implement local series data source for caching recent series search.

Mohamed-Ashraf-2210 committed:
feat: implement use cases for fetching series by genre and name, and add Season and SeriesGenre data classes

skaik-mo committed:
feat: Implement series remote data source with TMDB API

This commit introduces the `SeriesRemoteDataSource` and its implementation `TmdbSeriesApiRemoteDataSource` for fetching series data from The Movie Database (TMDB) API.

Key changes:
- Added `SeriesRemoteDataSource` interface defining methods for fetching series by name, genre, and retrieving genres.
- Implemented `TmdbSeriesApiRemoteDataSource` using Ktor for HTTP requests.
- Created `RequestBuilder` to encapsulate Ktor request logic and handle responses/errors.
- Defined `BaseRequest` for constructing API requests.
- Added DTOs for series (`SeriesDto`, `SeriesInfoDto`) and genres (`GenreDto`, `GenresResponse`).
- Included various `NetworkRequestException` types for error handling.
- Configured Ktor client and serialization in the `series` module's `build.gradle.kts`.
- Integrated `SeriesRemoteDataSource` into the DI graph in `dataModule.kt`.
- Added necessary Ktor and Kotlinx Serialization dependencies.

Mostafa-khaled21 committed:
feat: implement Room database with caching for series and search results

skaik-mo committed:
Remove gradle.properties from Git tracking

skaik-mo committed:
Refactor: Enhance series remote data source and DI setup

This commit refactors the `series` remote data source by:

- Simplifying the `RequestBuilder` to directly return the response body or throw an exception, removing the `ApiResult` wrapper.
- Making `RequestBuilder`, `BaseRequest` and `TmdbSeriesApiRemoteDataSource` internal.
- Removing unused exception classes from `NetworkRequestException.kt`.
- Introducing `SeriesResponse` and `GenresResponse` data classes.
- Updating `BaseRequest` to include default authorization and accept headers and retrieve the API key from `BuildConfig`.
- Creating `seriesDataModule` for Koin dependency injection, providing necessary dependencies for the series remote data source.
- Integrating `seriesDataModule` into the main application's Koin setup.
- Adding `gradle.properties` to `.gitignore`.
- Updating the Kotlin serialization plugin version.
- Adding `TMDB_API_KEY` to `buildConfigField` in `series/build.gradle.kts`.
- Removing series-related Koin definitions from the main `dataModule.kt` as they are now handled by `seriesDataModule`.

Mostafa-khaled21 committed:
validate seriesList before saving search results in SeriesRoomLocalDataSource

Mostafa-khaled21 committed:
refactor: improve seriesList validation before saving search results

Mostafa-khaled21 committed:
add caching for series genres with getCachedGenres and saveGenres methods

Mohamed-Ashraf-2210 committed:
feat: add use cases for searching series by genre and name, and update SeriesRepository interface

Mostafa-khaled21 committed:
feat: add caching for series by genre and improve genre cache validation

Mohamed-Ashraf-2210 committed:
Merge pull request #115 from Rome-Squad/feature/series-domain

Domain series

Mostafa-khaled21 committed:
refactor: rename keyword to name in search cache DAO and entity

Mostafa-khaled21 committed:
refactor: rename getCachedSeriesForKeyword to getCachedSeriesForName in data source

Mostafa-khaled21 committed:
feat: integrate series database entities and DAOs into CineVerseDatabase

doaaabdelnaser committed:
implement safe API and local call handling with custom exceptions

Mohamed-Ashraf-2210 committed:
feat: update series repository interface to return lists instead of results and add custom exception classes

Mohamed-Ashraf-2210 committed:
feat: update series repository to include store functionality and refactor use cases for better operator invocation

Mohamed-Ashraf-2210 committed:
Merge branch 'feature/series-domain' into feature/series

skaik-mo committed:
Merge branch 'develop' into feature/series-remote

* develop:
  increase use cases test coverage
  test all explore usecases
  move safe calling to the repo and handle rquest codes in the remote datasource
  move safe calling to the repo and handle rquest codes in the remote datasource
  Update test_coverage.yml
  fix gradle secrets
  fix secrets file name
  update gradle/ app to use secrets from secrets.properties
  move secrets to the app module
  accept the page as parameter for remote request
  update the usecases to use execute instead invoke operator function
  add secrets to github action
  Remove secret.properties from tracking
  remove test code from main activity
  implement explore feature initial use cases
  refactor di modules to be separated
  implement explore repository
  add remote explore datasource
  setup explore feature local search history cache
  add explore feature modules

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/java/com/giraffe/cineverseapp/CineVerseApp.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/databaseModule.kt

Mostafa-khaled21 committed:
enhance search caching with case-insensitive keyword handling

Mohamed-Ashraf-2210 committed:
feat: change releaseYear type to String in Season and Series data classes, and update SeriesException class hierarchy

Mohamed-Ashraf-2210 committed:
feat: add fake series function and implement tests for series use cases

Mohamed-Ashraf-2210 committed:
Merge pull request #131 from Rome-Squad/feature/series-domain

testing domain series

skaik-mo committed:
Merge branch 'feature/series' into feature/series-remote

* feature/series:
  feat: add fake series function and implement tests for series use cases
  feat: change releaseYear type to String in Season and Series data classes, and update SeriesException class hierarchy
  feat: update series repository to include store functionality and refactor use cases for better operator invocation
  feat: update series repository interface to return lists instead of results and add custom exception classes
  feat: add use cases for searching series by genre and name, and update SeriesRepository interface
  feat: implement use cases for fetching series by genre and name, and add Season and SeriesGenre data classes

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/feature/series' into feature/series-room

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/develop' into feature/series-room

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/java/com/giraffe/cineverseapp/data/database/CineVerseDatabase.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/databaseModule.kt

doaaabdelnaser committed:
Merge branch 'feature/series' into feature/series_safcall

Mostafa-khaled21 committed:
feat: refactor series data models and update database integration

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/develop' into feature/series

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/feature/series' into feature/series-room

doaaabdelnaser committed:
move and implement safe call handling with custom network and local exceptions from dataSource to repository

skaik-mo committed:
Refactor: Relocate series data source and models

This commit relocates the `SeriesRemoteDataSource` interface and the `SeriesDto` and `GenreDto` data classes from the `datasource/remote/series` module to the `repository/series` module.

Additionally, the following changes were made:

- Updated `TmdbSeriesApiRemoteDataSource` to use the new location of `SeriesRemoteDataSource` and the model classes.
- Modified `TmdbSeriesApiRemoteDataSource` to accept `genreId` as `Int` instead of `String`.
- Updated `BaseRequest` to accept `baseURL` and `accessToken` in its constructor and removed unused methods.
- Updated `build.gradle.kts` files to reflect the new module dependencies and plugin usage.
- Renamed Koin module from `seriesDataModule` to `seriesRemoteDataModule`.
- Adjusted Koin module to retrieve `BASE_URL` and `ACCESS_TOKEN` using named qualifiers.
- Removed explicit `@SerialName` annotations from DTOs where the property name matches the JSON key.
- Reordered fields in `SeriesDto`.

skaik-mo committed:
Merge branch 'feature/series' into feature/series-remote

* feature/series:

Mostafa-khaled21 committed:
Merge pull request #132 from Rome-Squad/feature/series-room

series room

skaik-mo committed:
Merge branch 'refs/heads/feature/series' into feature/series-remote

# Conflicts:
#	gradle/libs.versions.toml
#	repository/series/build.gradle.kts
#	repository/series/src/main/java/com/giraffe/series/model/SeriesDto.kt

skaik-mo committed:
Merge branch ' feature/series' into feature/series-remote

Mostafa-khaled21 committed:
fix: update search cache table name and queries for consistency

skaik-mo committed:
Merge branch 'feature/series' into feature/series-remote

* feature/series:
  fix: update search cache table name and queries for consistency

# Conflicts:
#	repository/series/src/main/java/com/giraffe/series/model/CachedSearchCacheDto.kt
#	repository/series/src/main/java/com/giraffe/series/util/DatabaseConstants.kt

Mohamed-Ashraf-2210 committed:
Merge pull request #116 from Rome-Squad/feature/series-remote

feat: Implement series remote data source with TMDB API

doaaabdelnaser committed:
rename and update exception mapping for safe call handling

doaaabdelnaser committed:
rename and update exception mapping for safe call handling

doaaabdelnaser committed:
rename and update exception mapping for safe call handling

doaaabdelnaser committed:
solve github conflict

doaaabdelnaser committed:
solve github conflict

doaaabdelnaser committed:
Merge branch 'feature/series' into feature/series_safcall

# Conflicts:
#	datasource/local/series/build.gradle.kts
#	datasource/remote/series/build.gradle.kts

Mostafa-khaled21 committed:
feat: update network module and refactor API request classes for improved integration

Mostafa-khaled21 committed:
feat: implement use cases for clearing and retrieving recent series

Mostafa-khaled21 committed:
feat: add recent series management with caching and repository integration

skaik-mo committed:
Merge pull request #127 from Rome-Squad/feature/series_safcall

 implement safe API and local call handling with custom exceptions

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/feature/series' into feature/series-repository

Mostafa-khaled21 committed:
feat: enhance exception handling and refactor series repository implementation

Mostafa-khaled21 committed:
feat: refactor series use cases and enhance test coverage with new exceptions

Mostafa-khaled21 committed:
Merge pull request #138 from Rome-Squad/feature/series-repository

series repository

skaik-mo committed:
Merge branch 'feature/series' into feature/series-remote

skaik-mo committed:
refactor: enhance network request handling and series data models

This commit introduces several improvements to the network request handling and data models within the `series` module.

**Key changes:**

- **Enhanced RequestBuilder:**
    - The `RequestBuilder` is now an interface, with `DefaultRequestBuilder` as its implementation.
    - Added comprehensive error handling to `DefaultRequestBuilder` to map Ktor exceptions and HTTP status codes to custom `NetworkException` types (e.g., `NoInternetNetworkException`, `ServerNetworkException`, `ClientException`). This provides more granular error information to the repository layer.
- **Updated SeriesDto and new SeasonDto:**
    - The `SeriesDto` has been updated to include `seasons` (a list of `SeasonDto`) and reordered some fields.
    - A new `SeasonDto` data class has been added to represent season information.
- **Refined SafeCall Utility:**
    - The `safeCall` utility function in `repository/series` has been updated to correctly map the new `NetworkException` types to `SeriesException` subtypes.
- **Improved DI for Series Data Source:**
    - The Koin module in `datasource/remote/series` now correctly provides `DefaultRequestBuilder` for the `RequestBuilder` interface.
    - Removed redundant Koin definitions for `BaseRequest`, `RequestBuilder`, and `SeriesRemoteDataSource` from the main `app` module as they are now handled within the `series` datasource module.
- **Unit Tests for Remote Data Source:**
    - Added unit tests for `TmdbSeriesApiRemoteDataSource` to verify its interaction with `RequestBuilder` and correct parsing of responses.
- **Removed Example Unit Test:**
    - Deleted the boilerplate `ExampleUnitTest.kt` file.

These changes improve the robustness of network operations, provide more detailed error information, and enhance the data modeling for TV series.

skaik-mo committed:
Merge branch 'develop' into feature/series

* develop:
  refactor: remove message parameter from PersonException constructor
  fix: add PersonDto to CineVerseDatabase entities
  test: add unit tests for PersonRepositoryImpl and SafeCall utility
  test: add unit tests for ClearRecentPeopleUseCase and GetRecentPeopleUseCase
  feat: add exception handling for empty results in GetRecentPeopleUseCase and SearchPeopleByNameUseCase
  feat: refactor person module structure and add ApiErrorResponse and ApiException classes
  refactor: remove API_KEY configuration from build.gradle.kts and clean up network module imports
  feat: implement person module with recent people use cases and data access layers
  feat: implement GetPeopleUseCase and add recent people retrieval to repository and datasource
  add domain tests and repo impl get person by name
  feat: add unit tests for GetPersonUseCase and refine exception handling
  feat: implement person repository, add person domain exceptions
  feat: add RequestBuilder for API requests and handle error responses
  move database and dao to app module
  feat: add remote data source for person module and update permissions
  delete dublicated refactor code
  implement local data source for person search with rome database
  feat: add remote data source for person search with Ktor integration
  feat: implement person module use cases

# Conflicts:
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/com/giraffe/cineverseapp/data/database/CineVerseDatabase.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/localDataSourceModule.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/repositoryModule.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/useCaseModule.kt
#	gradle/libs.versions.toml

HendSayed25 commented on repository/series/src/main/java/com/giraffe/series/model/CachedSearchCacheDto.kt:
```diff
@@ -0,0 +1,11 @@
+import com.giraffe.series.utils.DatabaseConstants.SEARCH_KEYWORD_TABLE_CACHE
+
+@Entity(tableName = SEARCH_KEYWORD_TABLE_CACHE)
+data class CachedSearchCacheDto(
```
What is the meaning of this name?
I think CachedSearchDto  or SearchCacheDto is readable more than this

skaik-mo committed:
Refactor: Remove gradle.properties from .gitignore

This commit removes `gradle.properties` from the `.gitignore` file.
It also introduces a new `gradle.properties` file with project-wide Gradle settings, including:
- JVM arguments for the daemon process.
- Enabling AndroidX.
- Setting Kotlin code style to "official".
- Enabling non-transitive R classes.

HendSayed25 commented on repository/series/src/main/java/com/giraffe/series/datasource/remote/SeriesRemoteDataSource.kt:
```diff
@@ -0,0 +1,10 @@
+import com.giraffe.series.model.SeriesDto
+
+interface SeriesRemoteDataSource {
+    suspend fun getSeriesByName(name: String, language: String = "en-US", includeAdult: Boolean = false, page: Int = 1): List<SeriesDto>
```
Why you take all these parameter and the name of function only refer to only name ?

I want to understand your idea..

mohamedebrahem13 replied:
the language will be used in the ktor interceptor. you don't need to direct pass it here  , and the include Adult you can use it in the ktor interceptor also it prevents you to pass it every time you call api

HendSayed25 commented on datasource/remote/series/src/test/java/com/giraffe/series/TmdbSeriesApiRemoteDataSourceTest.kt:
```diff
@@ -0,0 +1,266 @@
+        coVerify(exactly = 1) { requestBuilder.request(any()) }
+    }
+
+//    @Test
```
delete all these comments before merging

HendSayed25 commented on datasource/remote/series/src/main/java/com/giraffe/series/TmdbSeriesApiRemoteDataSource.kt:
```diff
@@ -0,0 +1,73 @@
+import io.ktor.http.HttpMethod
+
+
+class TmdbSeriesApiRemoteDataSource(
```
I think tmdb not readable for every one , 
SeriesRemoteDataSource is better

dev-bilal-azzam replied:
I think it is fine hend as long as this is an implementation of the data source so i think it is fine to have a name that refers to the implementation

HendSayed25 replied:
Tmdb is the name of api we used now but it may changed in any time 
I think using general name is better

skaik-mo replied:
When the API changes, we should create a new file with implementation instead of modifying the existing one.

HendSayed25 commented on app/src/main/java/com/giraffe/cineverseapp/MainActivity.kt:
```diff
@@ -19,8 +19,8 @@ val getKeywords: GetSearchKeywordsUseCase by inject()
                 LaunchedEffect(Unit) {
-                    val keywords = getKeywords.execute("Batman")
-                    Log.d("TAG", "onCreate: $keywords")
+//                    val keywords = getKeywords.execute("Batman")
```
Remove these comments

dev-bilal-azzam commented on datasource/local/series/src/main/java/com/giraffe/series/SeriesRoomLocalDateSource.kt:
```diff
@@ -0,0 +1,144 @@
+
+class SeriesRoomLocalDateSource(
+    private val seriesDao: SeriesDao,
+    private val cacheDao: SearchCacheDao
```
ambiguous names
series dao is also for cache
seriesDao amd searchHistoryDao is better

dev-bilal-azzam commented on app/src/main/java/com/giraffe/cineverseapp/data/database/CineVerseDatabase.kt:
```diff
@@ -3,23 +3,38 @@ package com.giraffe.cineverseapp.data.database
+        CachedSeriesDto::class,
+        CachedSeasonDto::class,
+        CachedSeriesGenreDto::class,
+        CachedSearchCacheDto::class,
```
Why this table exist?

dev-bilal-azzam replied:
there is already a table for the search history you just need to use it or retrieve data from it, not to manage it nor to isert or delete it, it is the explore(search) feature resposibility.

Mostafa-khaled21 replied:
to store keyword and last Searched Time

dev-bilal-azzam commented on datasource/local/series/src/main/java/com/giraffe/series/database/SearchCacheDao.kt:
```diff
@@ -0,0 +1,22 @@
+import androidx.room.*
+import com.giraffe.series.model.CachedSearchCacheDto
+@Dao
+interface SearchCacheDao {
```
series is not responsible for controlling search history, it just need to retrieve the search history to check last hour seacrches

dev-bilal-azzam commented on datasource/local/series/src/main/java/com/giraffe/series/database/SeriesDao.kt:
```diff
@@ -0,0 +1,63 @@
+    suspend fun insertSeries(series: List<CachedSeriesDto>)
+
+    @Query("SELECT * FROM cached_series")
+    fun getAllSeries(): Flow<List<CachedSeriesDto>>
```
why flow here I think you don't need to use flow just a list is enough

HendSayed25 commented on datasource/local/series/src/main/java/com/giraffe/series/database/SeriesDao.kt:
```diff
@@ -0,0 +1,63 @@
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertSeries(series: List<CachedSeriesDto>)
+
+    @Query("SELECT * FROM cached_series")
```
Use table name constant from database constant class instead you write it

dev-bilal-azzam commented on datasource/local/series/src/main/java/com/giraffe/series/database/SeriesDao.kt:
```diff
@@ -0,0 +1,63 @@
+    suspend fun insertSeasons(seasons: List<CachedSeasonDto>)
+
+    @Query("SELECT * FROM cached_season WHERE seriesId = :seriesId")
+    fun getSeasonsForSeries(seriesId: Int): Flow<List<CachedSeasonDto>>
```
the same as getAllSeries: unneedded flow

dev-bilal-azzam commented on datasource/remote/series/build.gradle.kts:
```diff
@@ -12,6 +13,12 @@ android {
 
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
         consumerProguardFiles("consumer-rules.pro")
+
```
no need for that the secrets (api_ky, access token , base url ) are already provided by app module using di and you just need to inject them wherever you need.

skaik-mo committed:
Remove commented-out code and unused imports

This commit removes unused code that was commented out in `TmdbSeriesApiRemoteDataSourceTest.kt` and `MainActivity.kt`. It also removes an unused import from `MainActivity.kt`.

dev-bilal-azzam commented on datasource/remote/series/src/main/java/com/giraffe/series/di/SeriesDataModule.kt:
```diff
@@ -0,0 +1,27 @@
+import org.koin.dsl.module
+
+
+val seriesRemoteDataModule = module {
```
there is already di module in the app so you don't need data source to depend on di framework

skaik-mo replied:
The classes are marked `internal` since they're module, it keeps the scope limited and makes refactoring easier

dev-bilal-azzam commented on repository/series/src/main/java/com/giraffe/series/model/SeriesDto.kt:
```diff
@@ -0,0 +1,89 @@
+
+
+@Entity(tableName = SERIES_TABLE)
+data class CachedSeriesDto(
```
where is the cached series genre?

Mostafa-khaled21 replied:
in the same file

dev-bilal-azzam commented on repository/series/src/main/java/com/giraffe/series/model/SeriesFullData.kt:
```diff
@@ -0,0 +1,8 @@
+package com.giraffe.series.model
+
+
+data class SeriesFullData(
```
why this model?

mohamedebrahem13 replied:
same question why you need extra model

Mostafa-khaled21 replied:
Replaced Triple with SeriesFullData to improve readability and maintainability — clearer property names and easier future extensions.

Mostafa-khaled21 committed:
feat: rename CachedSearchCacheDto to SearchCacheDto and update references

mohamedebrahem13 commented on repository/series/src/main/java/com/giraffe/series/datasource/remote/SeriesRemoteDataSource.kt:
```diff
@@ -0,0 +1,10 @@
+
+interface SeriesRemoteDataSource {
+    suspend fun getSeriesByName(name: String, language: String = "en-US", includeAdult: Boolean = false, page: Int = 1): List<SeriesDto>
+    suspend fun getSeriesByGenreId(genreId: Int, language: String = "en-US", includeAdult: Boolean = false, page: Int = 1, sortBy: String = "popularity.desc", includeNullFirstAirDates: Boolean = false): List<SeriesDto>
```
the language will be used in the ktor interceptor. you don't need to direct pass it here , and the include Adult you can use it in the ktor interceptor also it prevents you to pass it every time you call api

mohamedebrahem13 commented on repository/series/src/test/java/com/giraffe/series/SeriesRepositoryImplTest.kt:
```diff
@@ -0,0 +1,158 @@
+
+  val result = repository.searchSeriesByName("vikings")
+
+  assertEquals("Vikings", result.first().name)
```
we have truth lib from google for assertion

mohamedebrahem13 commented on repository/series/src/main/java/com/giraffe/series/model/SeriesDto.kt:
```diff
@@ -0,0 +1,89 @@
+    val rate: Float,
+    val duration: String,
+    val posterUrl: String,
+    val genresID: List<Int>,
```
did you add the type converter for this  val genresID: List<Int> 
room will not able to add it direct

Mostafa-khaled21 replied:
already added check CineVerseDatabase

mohamedebrahem13 commented on repository/series/src/main/java/com/giraffe/series/utils/safeCall.kt:
```diff
@@ -0,0 +1,61 @@
+        is InvalidRequestMethodException -> ValidationExceptions()
+        is InvalidRequestException -> ValidationExceptions()
+        is UnknownNetworkException -> UnknownException()
+
```
can you remove these extra space here ->

mohamedebrahem13 commented on repository/series/src/main/java/com/giraffe/series/datasource/remote/SeriesRemoteDataSource.kt:
```diff
@@ -0,0 +1,10 @@
+interface SeriesRemoteDataSource {
+    suspend fun getSeriesByName(name: String, language: String = "en-US", includeAdult: Boolean = false, page: Int = 1): List<SeriesDto>
+    suspend fun getSeriesByGenreId(genreId: Int, language: String = "en-US", includeAdult: Boolean = false, page: Int = 1, sortBy: String = "popularity.desc", includeNullFirstAirDates: Boolean = false): List<SeriesDto>
+    suspend fun getGenres(language: String = "en-US"): List<GenreDto>
```
don't pass the language here you can use it direct at the ktor  part

mohamedebrahem13 commented on repository/series/src/main/java/com/giraffe/series/datasource/local/SeriesLocalDateSource.kt:
```diff
@@ -0,0 +1,29 @@
+    suspend fun getCachedGenres(): List<CachedSeriesGenreDto>
+    suspend fun saveGenres(genres: List<CachedSeriesGenreDto>)
+
+    suspend fun getCachedSeriesByGenre(genreId: Int): List<SeriesFullData>?
```
remove the null you can return empty of there is no data cashed and then check if empty and get from remote

mohamedebrahem13 commented on repository/series/src/main/java/com/giraffe/series/datasource/local/SeriesLocalDateSource.kt:
```diff
@@ -0,0 +1,29 @@
+
+interface SeriesLocalDateSource {
+
+    suspend fun getCachedSeriesForName(name: String): List<SeriesFullData>?
```
remove the null you can return empty if there is no data cashed and then check if empty and get from remote

Mostafa-khaled21 committed:
feat: rename SearchCacheDao to searchHistoryDao and update references

Mostafa-khaled21 committed:
feat: update return types in series data source methods for consistency

Mostafa-khaled21 committed:
feat: add Truth assertions in tests and update dependencies for consistency

skaik-mo committed:
Refactor: Remove default parameters from SeriesRemoteDataSource

This commit simplifies the `SeriesRemoteDataSource` interface by removing default parameter values for `language`, `includeAdult`, `sortBy`, and `includeNullFirstAirDates` from its methods.

The corresponding `TmdbSeriesApiRemoteDataSource` implementation has been updated to reflect these changes.

Additionally, tests in `TmdbSeriesApiRemoteDataSourceTest` have been adjusted to align with the modified interface.

Mostafa-khaled21 committed:
feat: remove unused imports and clean up database-related code

skaik-mo committed:
Refactor: Removed TMDB_API_KEY from the `datasource/remote/series build gradle

The TMDB_API_KEY build configuration field has been removed from the `datasource/remote/series` module. This change is likely part of a refactor to centralize API key management in the main application module.

Mostafa-khaled21 committed:
feat: update return type for getCachedSeriesForName and clean up method signatures

Mostafa-khaled21 committed:
feat: remove unused parameters and clean up series data source methods

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/feature/series' into feature/series

Mostafa-khaled21 committed:
feat: refactor SeriesRoomLocalDateSourceTest to improve clarity and coverage

Mostafa-khaled21 committed:
feat: remove unused duration parameter from Series and update related tests

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/develop' into feature/series

# Conflicts:
#	app/src/main/java/com/giraffe/cineverseapp/MainActivity.kt
#	app/src/main/java/com/giraffe/cineverseapp/data/database/CineVerseDatabase.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/localDataSourceModule.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/repositoryModule.kt
#	app/src/main/java/com/giraffe/cineverseapp/di/useCaseModule.kt
#	gradle/libs.versions.toml

skaik-mo committed:
Merge branch 'develop' into feature/explore-screen

# Conflicts:
#	app/src/main/java/com/giraffe/cineverseapp/data/database/CineVerseDatabase.kt
